src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -716,12 +719,19 @@ def build_vmss_resource(name, naming_prefix, location, tags, overprovision, upgr
             }
         }
     }
+
+    vmss_api_version = '2016-04-30-preview'
+    if get_api_version(ResourceType.MGMT_COMPUTE) in ['2016-04-30-preview']:
+        vmss_properties.update({'singlePlacementGroup': single_placement_group})
+    else:
+        vmss_api_version='2016-03-30'
+
     vmss = {
         'type': 'Microsoft.Compute/virtualMachineScaleSets',
         'name': name,
         'location': location,
         'tags': tags,
-        'apiVersion': '2016-04-30-preview',
+        'apiVersion': vmss_api_version,
```

Comment:
Should get the API version from `get_api_version` like you do for `vm_api_version` above.